doc/services/pm/device.rst

```diff
@@ -158,8 +157,54 @@ a particular device. It is important to emphasize that, although the
 state is tracked by the subsystem, it is the responsibility of each device driver
 to handle device actions(:c:enum:`pm_device_action`) which change device state.
 
-Each :c:enum:`pm_device_action` have a direct an unambiguous relationship with
-a :c:enum:`pm_device_state`.
+Each :c:enum:`pm_device_state` state is defined as follows:
+
+- ``OFF``: Device hardware is not powered. This is the initial state from which
+  a device driver is initialized.
+
+  When a device driver is initialized, we do not know the state of the device.
+  As a result, the :c:enum:`pm_device_action` ``TURN_ON`` action should be able
+  to transition the device from any internal state into ``SUSPENDED``, since no
+  guarantees can be made across resets. This is typically achieved through
+  toggling a reset pin or triggering a software reset through a register write
+  before performing any additional configuration needed to meet the requirements
+  of ``SUSPENDED``. For devices where this is not possible, the device driver
+  must presume the device is in either the ``OFF`` or ``SUSPENDED`` state at
+  time of initialization, as these are the states within which device drivers
+  may be deinitialized.
+
+- ``SUSPENDED``: Device hardware is powered, but the device is not needed by the
+  system. The device should be put into its lowest internal power state,
+  commonly named "disabled" or "stopped". The device will be powered off if its
+  power domain is no longer needed by the system. If the device itself is a
+  power domain, devices on this power domain will be powered off in this state.
+
+  Note that this state is **NOT** a "low-power"/"partially operable" state,
+  those are configured using device driver specific APIs, and apply only while
+  the device is ``ACTIVE``.
+
+  A device driver may be deinitialized in this state. Once the device driver
+  has been deinitialized, we implicitly move to the ``OFF`` state as the device
+  hardware may lose power, with no device driver to respond to the
+  corresponding :c:enum:`pm_device_action` ``TURN_OFF`` action.
+
+- ``ACTIVE``: Device hardware is powered, and the device is needed by the
+  system. The device should be enabled in this state. Any device driver API may
+  be called in this state.
+
+- ``SUSPENDING``: Device hardware is powered, but the device has been scheduled
+  to be suspended, as it is no longer needed by the system. This state is used
+  when delegating suspension of a device to the PM subsystem, optionally with
+  residency to avoid unnecessary suspend/resume cycles. The device will be
```

Comment:
The wording is confusing to me here. Is it saying the residency is optional or this "delegated suspension" is optional and would always have residency.

Also, I am maybe lacking knowledge here, (perhaps that makes me qualified to review the documentation though), but I am not understanding how a device would get into the suspending state as opposed to going directly to suspend. And this residency that is mentioned, is it referring to the system mode residency or is there some other value associated for the device that is configured somehow?

Reply:
I think what you mean is that it is not mapped to a device *action* (as opposed to state). The actions are how the device driver makes sense of what it is supposed to do, and as you have pointed out, this state (suspending) does not seem to require any action on the part of a device.